operator/v1alpha1/types_crdchecker.go

```diff
@@ -0,0 +1,346 @@
+	// When not specified, the field is excluded from all versions.
+	// When present, must be a valid Kubernetes API version string, with a
+	// maximum length of 255 characters.
+	// +optional
+	Version APIVersionString `json:"version,omitempty"`
+}
+
+// CompatibilitySchema defines the schema used by crdSchemaValidation and objectSchemaValidation.
+type CompatibilitySchema struct {
+	// customResourceDefinition contains the complete definition of the CRD for schema and object validation purposes.
+	// This field is required.
+	// +required
+	CustomResourceDefinition CRDData `json:"customResourceDefinition,omitzero"`
+
+	// requiredVersions specifies a subset of the CRD's API versions which will be asserted for compatibility.
+	// This field is required.
+	// +required
+	RequiredVersions APIVersions `json:"requiredVersions,omitzero"`
+
+	// excludedFields is a set of fields in the schema which will not be validated by
+	// crdSchemaValidation or objectSchemaValidation.
+	// The list may contain at most 64 fields.
+	// When not specified, all fields in the schema will be validated.
+	// +kubebuilder:validation:MaxItems=64
+	// +listType=atomic
+	// +optional
+	ExcludedFields []APIExcludedField `json:"excludedFields,omitempty"`
+}
+
+// CustomResourceDefinitionSchemaValidation ensures that updates to the installed CRD are compatible with this compatibility requirement.
+type CustomResourceDefinitionSchemaValidation struct {
+	// action determines whether violations are rejected (Deny) or admitted with an API warning (Warn).
+	// Valid options are:
+	//   Deny - incompatible CRDs will be rejected and not admitted to the cluster.
+	//   Warn - incompatible CRDs will be allowed but a warning will be generated in the API response.
+	// This field is required.
+	// +required
+	Action CRDAdmitAction `json:"action,omitempty"`
+}
+
+// ObjectSchemaValidation ensures that matching objects conform to the compatibilitySchema.
+type ObjectSchemaValidation struct {
+	// action determines whether violations are rejected (Deny) or admitted with an API warning (Warn).
+	// Valid options are:
+	//   Deny - incompatible Objects will be rejected and not admitted to the cluster.
+	//   Warn - incompatible Objects will be allowed but a warning will be generated in the API response.
+	// This field is required.
+	// +required
+	Action CRDAdmitAction `json:"action,omitempty"`
+
+	// namespaceSelector defines a label selector for namespaces. If defined,
+	// only objects in a namespace with matching labels will be subject to
+	// validation. When not specified, objects for validation will not be
+	// filtered by namespace.
+	// +optional
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// objectSelector defines a label selector for objects. If defined, only
+	// objects with matching labels will be subject to validation. When not
+	// specified, objects for validation will not be filtered by label.
```

Comment:
```suggestion
	// Default to the empty LabelSelector, which matches everything.
```

From ValidatingWebhookConfiguration (which is way more verbose)